### PR TITLE
Initial Azure.ResourceManager.DataFactory

### LINKED ISSUE #29333: Initial Azure.ResourceManager.DataFactory
https://github.com/Azure/azure-rest-api-specs/blob/de400f7204d30d25543ac967636180728d52a88f/specification/datafactory/resource-manager/readme.md

### PR DESCRIPTION
**Pull request by Yao725**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: eng/CredScanSuppression.json [hunk 1]
```diff
@@ -90,6 +90,12 @@
                 "sdk/keyvault/Azure.Security.KeyVault.Certificates/tests/SessionRecords/CertificateClientLiveTests/VerifyImportCertificatePemWithoutIssuerAsync.json"
             ],
             "_justification": "Files contain ephemeral keys or certificates used by test code during playback."
+        },
+        {
+            "file":[
+                "sdk/datafactory/Azure.ResourceManager.DataFactory/src/Generated/Models/RestServiceAuthenticationType.cs"
+            ],
+            "_justification": "File contains a string value that ends with credential, not a true credential."
```
Comment (Yao725) [inline eng/CredScanSuppression.json:98]: [CredScan](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1642909&view=logs&j=bc67675d-56bf-581f-e0a2-208848ba68ca&t=7eee3a58-6120-518b-7fcb-7e943712aa81&l=59) failed due to [this file](https://github.com/Azure/azure-sdk-for-net/blob/f06129c225bd15d05ad121871a27f03e6aac4cec/sdk/datafactory/Azure.ResourceManager.DataFactory/src/Generated/Models/RestServiceAuthenticationType.cs#L29), but it's a string value of the enum, not a true secret key.

File: sdk/datafactory/Azure.ResourceManager.DataFactory/src/Generated/Models/SsisObjectMetadataStatusResponse.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+namespace Azure.ResourceManager.DataFactory.Models
+{
+    /// <summary> The status of the operation. </summary>
+    public partial class SsisObjectMetadataStatusResponse
```
Comment (Yao725) [inline sdk/datafactory/Azure.ResourceManager.DataFactory/src/Generated/Models/SsisObjectMetadataStatusResponse.cs:11]:
```suggestion
    public partial class SsisObjectMetadataStatusResult
```

File: sdk/datafactory/Azure.ResourceManager.DataFactory/src/Generated/PipelineResource.cs [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+
+namespace Azure.ResourceManager.DataFactory
+{
+    /// <summary>
+    /// A Class representing a PipelineResource along with the instance operations that can be performed on it.
+    /// If you have a <see cref="ResourceIdentifier" /> you can construct a <see cref="PipelineResource" />
+    /// from an instance of <see cref="ArmClient" /> using the GetPipelineResource method.
+    /// Otherwise you can get one from its parent resource <see cref="FactoryResource" /> using the GetPipelineResource method.
+    /// </summary>
+    public partial class PipelineResource : ArmResource
```
Comment (Yao725) [inline sdk/datafactory/Azure.ResourceManager.DataFactory/src/Generated/PipelineResource.cs:27]: Need renaming.

File: sdk/datafactory/Azure.ResourceManager.DataFactory/src/Generated/Models/Trigger.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+using Azure.Core;
+
+namespace Azure.ResourceManager.DataFactory.Models
+{
+    /// <summary>
+    /// Azure data factory nested object which contains information about creating pipeline run
+    /// Please note <see cref="Trigger"/> is the base class. According to the scenario, a derived class of the base class might need to be assigned here, or this property needs to be casted to one of the possible derived classes.
+    /// The available derived classes include <see cref="BlobEventsTrigger"/>, <see cref="BlobTrigger"/>, <see cref="ChainingTrigger"/>, <see cref="CustomEventsTrigger"/>, <see cref="MultiplePipelineTrigger"/>, <see cref="RerunTumblingWindowTrigger"/>, <see cref="ScheduleTrigger"/> and <see cref="TumblingWindowTrigger"/>.
+    /// </summary>
+    public partial class Trigger
```
Comment (Yao725) [inline sdk/datafactory/Azure.ResourceManager.DataFactory/src/Generated/Models/Trigger.cs:19]: Single word class name.

**Review by Yao725 — COMMENTED** [review]
The biggest problem is that there are so many properties are incorrectly generated as Uri type.

File: sdk/datafactory/Azure.ResourceManager.DataFactory/src/Properties/AssemblyInfo.cs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Runtime.CompilerServices;
+
+[assembly: Azure.Core.AzureResourceProviderNamespace("DataFactory")]
```
Comment (ArthurMa1978) [inline sdk/datafactory/Azure.ResourceManager.DataFactory/src/Properties/AssemblyInfo.cs:6]: Microsoft.DataFactory

File: sdk/datafactory/Azure.ResourceManager.DataFactory/src/autorest.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+library-name: DataFactory
+namespace: Azure.ResourceManager.DataFactory
+require: https://github.com/Azure/azure-rest-api-specs/blob/de400f7204d30d25543ac967636180728d52a88f/specification/datafactory/resource-manager/readme.md
+output-folder: $(this-folder)/Generated
+clear-output-folder: true
+skip-csproj: true
+ 
+format-by-name-rules:
+  'tenantId': 'uuid'
+  'etag': 'etag'
```
Comment (ArthurMa1978) [inline sdk/datafactory/Azure.ResourceManager.DataFactory/src/autorest.md:18]: this is the old template, here should be 
‘ETag': 'etag'

File: sdk/datafactory/Azure.ResourceManager.DataFactory/src/autorest.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  DataFlowDebugSession_QueryByFactory: GetDataFlowDebugSessions
+  ExposureControl_QueryFeatureValuesByFactory: GetExposureControlFeatureValues
+  Triggers_QueryByFactory: GetTriggers
+  Factories_ConfigureFactoryRepo: ConfigureFactoryRepo
+
+directive:
+  - from: datafactory.json
+    where: $.definitions
+    transform: >
+      $.PurviewConfiguration.properties.purviewResourceId['x-ms-format'] = 'arm-id';
```
Comment (ArthurMa1978) [inline sdk/datafactory/Azure.ResourceManager.DataFactory/src/autorest.md:84]: Can use `format-by-name-rules` if this name `purviewResourceId` is unique or all properties has this name are `arm-id`

**Review by ArthurMa1978 — APPROVED** [review]

File: sdk/datafactory/Azure.ResourceManager.DataFactory/src/autorest.md (hunk 7 above)
Comment (Yao725) [reply sdk/datafactory/Azure.ResourceManager.DataFactory/src/autorest.md]: Fixed, `PurviewResourceId` is unique.

**Pull request merged**